tests/unit/test_router.py

```diff
@@ -99,6 +147,57 @@ def test_get_next_phase(current_phase, exp_next_phase, router_inst, event_msg,
     assert 1 == len(caplog.records)
 
 
+@pytest.mark.asyncio
+@pytest.mark.parametrize('exists,exp_start_count', (
+    (True, 0),
+    (False, 1),
+))
+async def test_add_message_in_flight(exists, exp_start_count, router_inst,
```

Comment:
I'm a little unclear on what `exists` is here.  Is it if the next plugin exists?

Reply:
It's actually if the event_msg exists (in the dict already). Would `msg_exists` work?